Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftWindowsEventFirewall.yaml

```diff
@@ -96,56 +99,68 @@ ParserQuery: |
-                              '\x0d\x0a  <Data Name="DestAddress">'DestAddress'</Data>'
-                              '\x0d\x0a  <Data Name="DestPort">'DestPort:int'</Data>'
-                              '\x0d\x0a  <Data Name="Protocol">'Protocol:int'</Data>'
-                              '\x0d\x0a  <Data Name="FilterRTID">'NetworkRuleNumber:int'</Data>'
-                              '\x0d\x0a  <Data Name="LayerName">'LayerCode'</Data>'
-                              '\x0d\x0a  <Data Name="LayerRTID">'LayerRTID'</Data>'*
+          | parse EventData with * 
+              '<Data Name="ProcessId">'ProcessId:string'</Data>'
+              '\x0d\x0a  <Data Name="Application">'Application'</Data>'
+              '\x0d\x0a  <Data Name="Direction">'DirectionCode'</Data>'
+              '\x0d\x0a  <Data Name="SourceAddress">'SourceAddress'</Data>'
+              '\x0d\x0a  <Data Name="SourcePort">'SourcePort:int'</Data>'
+              '\x0d\x0a  <Data Name="DestAddress">'DestAddress'</Data>'
+              '\x0d\x0a  <Data Name="DestPort">'DestPort:int'</Data>'
+              '\x0d\x0a  <Data Name="Protocol">'Protocol:int'</Data>'
+              '\x0d\x0a  <Data Name="FilterRTID">'NetworkRuleNumber:int'</Data>'
+              '\x0d\x0a  <Data Name="LayerName">'LayerCode'</Data>'
+              '\x0d\x0a  <Data Name="LayerRTID">'LayerRTID'</Data>'*
+          | extend temp_isSrcMatch=has_any_ipv4_prefix(SourceAddress,src_or_any)
+          , temp_isDstMatch=has_any_ipv4_prefix(DestAddress,dst_or_any)
+          | extend ASimMatchingIpAddr = case(
+            array_length(src_or_any) == 0 and array_length(dst_or_any) == 0, "-" // match not requested: probably most common case
+            , (temp_isSrcMatch and temp_isDstMatch), "Both" // has to be checked before the individual 
+            , temp_isSrcMatch, "SrcIpAddr"
+            , temp_isDstMatch, "DstIpAddr"
+            , "No match"
+            )
+          | project-away temp_*  
         ;
       let SecurityEvent_5154_5155_5158_5159 =
       SecurityEvent | where not(disabled)
           | where EventID in (5154, 5155, 5158, 5159)
           // ***************  Prefilterring *****************************************************************
-          |where (array_length(dstipaddr_has_any_prefix)==0 ) 
+          |where (isnull(starttime) or TimeGenerated>=starttime)
+            and  (isnull(endtime) or TimeGenerated<=endtime)
+            and  (array_length(dstipaddr_has_any_prefix)==0 ) 
             and  (array_length(hostname_has_any)==0 ) 
             and  (isnull(dstportnumber) ) 
-            and  (isnull(starttime) or TimeGenerated>=starttime)
-            and  (isnull(endtime) or TimeGenerated<=endtime)
-            and  (array_length(srcipaddr_has_any_prefix)==0 
-                  or has_any_ipv4_prefix(EventData ,srcipaddr_has_any_prefix)
+            and  (array_length(ip_any)==0 
+                  or has_any_ipv4_prefix(EventData ,ip_any)
```

Comment:
Should really be src_or_any

Reply:
I don't think so. This is an early filter that checks if ANY of the addresses are present in the data